---
layout: default
title: Samtools man page 
---
<h1 align=center>samtools</h1>

<a href="#NAME">NAME</a><br>
<a href="#SYNOPSIS">SYNOPSIS</a><br>
<a href="#DESCRIPTION">DESCRIPTION</a><br>
<a href="#COMMANDS AND OPTIONS">COMMANDS AND OPTIONS</a><br>
<a href="#REFERENCE SEQUENCES">REFERENCE SEQUENCES</a><br>
<a href="#ENVIRONMENT VARIABLES">ENVIRONMENT VARIABLES</a><br>
<a href="#EXAMPLES">EXAMPLES</a><br>
<a href="#LIMITATIONS">LIMITATIONS</a><br>
<a href="#AUTHOR">AUTHOR</a><br>
<a href="#SEE ALSO">SEE ALSO</a><br>

<hr>


<a name="NAME"></a>
<h2>NAME</h2>


<p style="margin-left:11%; margin-top: 1em">samtools
&minus; Utilities for the Sequence Alignment/Map (SAM)
format</p>

<a name="SYNOPSIS"></a>
<h2>SYNOPSIS</h2>


<p style="margin-left:11%; margin-top: 1em">samtools view
-bt ref_list.txt -o aln.bam aln.sam.gz</p>

<p style="margin-left:11%; margin-top: 1em">samtools sort
-T /tmp/aln.sorted -o aln.sorted.bam aln.bam</p>

<p style="margin-left:11%; margin-top: 1em">samtools index
aln.sorted.bam</p>

<p style="margin-left:11%; margin-top: 1em">samtools
idxstats aln.sorted.bam</p>

<p style="margin-left:11%; margin-top: 1em">samtools view
aln.sorted.bam chr2:20,100,000-20,200,000</p>

<p style="margin-left:11%; margin-top: 1em">samtools merge
out.bam in1.bam in2.bam in3.bam</p>

<p style="margin-left:11%; margin-top: 1em">samtools faidx
ref.fasta</p>

<p style="margin-left:11%; margin-top: 1em">samtools
mpileup -C50 -gf ref.fasta -r chr3:1,000-2,000 in1.bam
in2.bam</p>

<p style="margin-left:11%; margin-top: 1em">samtools tview
aln.sorted.bam ref.fasta</p>

<p style="margin-left:11%; margin-top: 1em">samtools flags
PAIRED,UNMAP,MUNMAP</p>

<p style="margin-left:11%; margin-top: 1em">samtools bam2fq
input.bam &gt; output.fastq</p>

<a name="DESCRIPTION"></a>
<h2>DESCRIPTION</h2>


<p style="margin-left:11%; margin-top: 1em">Samtools is a
set of utilities that manipulate alignments in the BAM
format. It imports from and exports to the SAM (Sequence
Alignment/Map) format, does sorting, merging and indexing,
and allows to retrieve reads in any regions swiftly.</p>

<p style="margin-left:11%; margin-top: 1em">Samtools is
designed to work on a stream. It regards an input file
&lsquo;-&rsquo; as the standard input (stdin) and an output
file &lsquo;-&rsquo; as the standard output (stdout).
Several commands can thus be combined with Unix pipes.
Samtools always output warning and error messages to the
standard error output (stderr).</p>

<p style="margin-left:11%; margin-top: 1em">Samtools is
also able to open a BAM (not SAM) file on a remote FTP or
HTTP server if the BAM file name starts with
&lsquo;ftp://&rsquo; or &lsquo;http://&rsquo;. Samtools
checks the current working directory for the index file and
will download the index upon absence. Samtools does not
retrieve the entire alignment file unless it is asked to do
so.</p>

<a name="COMMANDS AND OPTIONS"></a>
<h2>COMMANDS AND OPTIONS</h2>


<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="6%">


<p style="margin-top: 1em" valign="top"><b>view</b></p></td>
<td width="9%"></td>
<td width="74%">


<p style="margin-top: 1em" valign="top">samtools view
[<i>options</i>] <i>in.bam</i>|<i>in.sam</i>|<i>in.cram</i>
[<i>region</i>...]</p> </td>
</table>

<p style="margin-left:26%; margin-top: 1em">With no options
or regions specified, prints all alignments in the specified
input alignment file (in SAM, BAM, or CRAM format) to
standard output in SAM format (with no header).</p>

<p style="margin-left:26%; margin-top: 1em">You may specify
one or more space-separated region specifications after the
input filename to restrict output to only those alignments
which overlap the specified region(s). Use of region
specifications requires a coordinate-sorted and indexed
input file (in BAM or CRAM format).</p>

<p style="margin-left:26%; margin-top: 1em">The <b>-b</b>,
<b>-C</b>, <b>-1</b>, <b>-u</b>, <b>-h</b>, <b>-H</b>, and
<b>-c</b> options change the output format from the default
of headerless SAM, and the <b>-o</b> and <b>-U</b> options
set the output file name(s).</p>

<p style="margin-left:26%; margin-top: 1em">The <b>-t</b>
and <b>-T</b> options provide additional reference data. One
of these two options is required when SAM input does not
contain @SQ headers, and the <b>-T</b> option is required
whenever writing CRAM output.</p>

<p style="margin-left:26%; margin-top: 1em">The <b>-L</b>,
<b>-r</b>, <b>-R</b>, <b>-q</b>, <b>-l</b>, <b>-m</b>,
<b>-f</b>, and <b>-F</b> options filter the alignments that
will be included in the output to only those alignments that
match certain criteria.</p>

<p style="margin-left:26%; margin-top: 1em">The <b>-x</b>,
<b>-B</b>, <b>-s</b>, and <b>-Q</b> options modify the data
which is contained in each alignment.</p>

<p style="margin-left:26%; margin-top: 1em">Finally, the
<b>-@</b> option can be used to allocate additional threads
to be used for compression, and the <b>-?</b> option
requests a long help message.</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="12%">



<p style="margin-top: 1em" valign="top"><b>REGIONS:</b></p> </td>
<td width="3%"></td>
<td width="74%">


<p style="margin-top: 1em" valign="top">Regions can be
specified as: RNAME[:STARTPOS[-ENDPOS]] and all position
coordinates are 1-based.</p></td>
</table>

<p style="margin-left:26%; margin-top: 1em">Important note:
when multiple regions are given, some alignments may be
output multiple times if they overlap more than one of the
specified regions.</p>

<p style="margin-left:26%; margin-top: 1em">Examples of
region specifications:</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="26%"></td>
<td width="9%">



<p style="margin-top: 1em" valign="top">&lsquo;chr1&rsquo;</p> </td>
<td width="7%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Output all
alignments mapped to the reference sequence named
&lsquo;chr1&rsquo; (i.e. @SQ SN:chr1) .</p></td>
</table>

<p style="margin-left:26%;">&lsquo;chr2:1000000&rsquo;</p>

<p style="margin-left:42%;">The region on chr2 beginning at
base position 1,000,000 and ending at the end of the
chromosome.</p>


<p style="margin-left:26%;">&lsquo;chr3:1000-2000&rsquo;</p>

<p style="margin-left:42%;">The 1001bp region on chr3
beginning at base position 1,000 and ending at base position
2,000 (including both end positions).</p>


<p style="margin-left:11%; margin-top: 1em"><b>OPTIONS:</b></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-b</b></p></td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Output in the BAM
format.</p> </td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-C</b></p></td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Output in the CRAM
format (requires -T).</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-1</b></p></td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Enable fast BAM
compression (implies -b).</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-u</b></p></td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Output uncompressed
BAM. This option saves time spent on
compression/decompression and is thus preferred when the
output is piped to another samtools command.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-h</b></p></td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Include the header
in the output.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-H</b></p></td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Output the header
only.</p> </td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-c</b></p></td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Instead of printing
the alignments, only count them and print the total number.
All filter options, such as <b>-f</b>, <b>-F</b>, and
<b>-q</b>, are taken into account.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-?</b></p></td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Output long help
and exit immediately.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-o</b>
<i>FILE</i></p> </td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Output to <i>FILE
[stdout].</i></p> </td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-U</b>
<i>FILE</i></p> </td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Write alignments
that are <i>not</i> selected by the various filter options
to <i>FILE</i>. When this option is used, all alignments (or
all alignments intersecting the <i>regions</i> specified)
are written to either the output file or this file, but
never both.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-t</b>
<i>FILE</i></p> </td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">A tab-delimited
<i>FILE</i>. Each line must contain the reference name in
the first column and the length of the reference in the
second column, with one line for each distinct reference.
Any additional fields beyond the second column are ignored.
This file also defines the order of the reference sequences
in sorting. If you run: &lsquo;samtools faidx
&lt;ref.fa&gt;&rsquo;, the resulting index file
<i>&lt;ref.fa&gt;.fai</i> can be used as this
<i>FILE</i>.</p> </td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-T</b>
<i>FILE</i></p> </td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">A FASTA format
reference <i>FILE</i>, optionally compressed by <b>bgzip</b>
and ideally indexed by <b>samtools faidx</b>. If an index is
not present, one will be generated for you.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-L</b>
<i>FILE</i></p> </td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Only output
alignments overlapping the input BED <i>FILE</i> [null].</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-r</b>
<i>STR</i></p> </td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Only output
alignments in read group <i>STR</i> [null].</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-R</b>
<i>FILE</i></p> </td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Output alignments
in read groups listed in <i>FILE</i> [null].</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-q</b>
<i>INT</i></p> </td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Skip alignments
with MAPQ smaller than <i>INT</i> [0].</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-l</b>
<i>STR</i></p> </td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Only output
alignments in library <i>STR</i> [null].</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-m</b>
<i>INT</i></p> </td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Only output
alignments with number of CIGAR bases consuming query
sequence &gt;= <i>INT</i> [0]</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-f</b>
<i>INT</i></p> </td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Only output
alignments with all bits set in <i>INT</i> present in the
FLAG field. <i>INT</i> can be specified in hex by beginning
with &lsquo;0x&rsquo; (i.e. /^0x[0-9A-F]+/) or in octal by
beginning with &lsquo;0&rsquo; (i.e. /^0[0-7]+/) [0].</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-F</b>
<i>INT</i></p> </td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Do not output
alignments with any bits set in <i>INT</i> present in the
FLAG field. <i>INT</i> can be specified in hex by beginning
with &lsquo;0x&rsquo; (i.e. /^0x[0-9A-F]+/) or in octal by
beginning with &lsquo;0&rsquo; (i.e. /^0[0-7]+/) [0].</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-x</b>
<i>STR</i></p> </td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Read tag to exclude
from output (repeatable) [null]</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-B</b></p></td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Collapse the
backward CIGAR operation.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-s</b>
<i>FLOAT</i></p> </td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Integer part is
used to seed the random number generator [0]. Part after the
decimal point sets the fraction of templates/pairs to
subsample [no subsampling].</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-Q</b>
<i>INT</i></p> </td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Scale quality
values in output by INT [1]</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-@</b>
<i>INT</i></p> </td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Number of BAM
compression threads to use in addition to main thread
[0].</p> </td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p style="margin-top: 1em" valign="top"><b>-S</b></p></td>
<td width="4%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Ignored for
compatibility with previous samtools versions. Previously
this option was required if input was in SAM format, but now
the correct format is automatically detected by examining
the first few characters of input.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="27%">


<p><b>tview</b></p></td>
<td width="4%"></td>
<td width="58%">
</td>
</table>

<p style="margin-left:11%;">samtools tview [<b>&minus;p</b>
<i>chr:pos</i>] [<b>&minus;s</b> <i>STR</i>]
[<b>&minus;d</b> <i>display</i>] &lt;in.sorted.bam&gt;
[ref.fasta]</p>

<p style="margin-left:26%; margin-top: 1em">Text alignment
viewer (based on the ncurses library). In the viewer, press
&lsquo;?&rsquo; for help and press &lsquo;g&rsquo; to check
the alignment start from a region in the format like
&lsquo;chr10:10,000,000&rsquo; or &lsquo;=10,000,000&rsquo;
when viewing the same reference sequence.</p>


<p style="margin-left:26%; margin-top: 1em"><b>Options:</b></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="31%">



<p style="margin-top: 1em" valign="top"><b>-d&nbsp;</b><i>display</i></p> </td>
<td width="6%"></td>
<td width="52%">


<p style="margin-top: 1em" valign="top">Output as (H)tml or
(C)urses or (T)ext</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="31%">



<p style="margin-top: 1em" valign="top"><b>-p&nbsp;</b><i>chr:pos</i></p> </td>
<td width="6%"></td>
<td width="52%">


<p style="margin-top: 1em" valign="top">Go directly to this
position</p> </td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="31%">



<p style="margin-top: 1em" valign="top"><b>-s&nbsp;</b><i>STR</i></p> </td>
<td width="6%"></td>
<td width="52%">


<p style="margin-top: 1em" valign="top">Display only
alignments from this sample or read group</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="31%">


<p><b>mpileup</b></p></td>
<td width="6%"></td>
<td width="52%">
</td>
</table>

<p style="margin-left:11%;">samtools mpileup
[<b>&minus;EBugp</b>] [<b>&minus;C</b> <i>capQcoef</i>]
[<b>&minus;r</b> <i>reg</i>] [<b>&minus;f</b> <i>in.fa</i>]
[<b>&minus;l</b> <i>list</i>] [<b>&minus;Q</b>
<i>minBaseQ</i>] [<b>&minus;q</b> <i>minMapQ</i>]
<i>in.bam</i> [<i>in2.bam</i> [<i>...</i>]]</p>

<p style="margin-left:26%; margin-top: 1em">Generate VCF,
BCF or pileup for one or multiple BAM files. Alignment
records are grouped by sample (SM) identifiers in @RG header
lines. If sample identifiers are absent, each input file is
regarded as one sample.</p>

<p style="margin-left:26%; margin-top: 1em">In the pileup
format (without <b>-u&nbsp;</b>or&nbsp;<b>-g</b>), each line
represents a genomic position, consisting of chromosome
name, 1-based coordinate, reference base, the number of
reads covering the site, read bases, base qualities and
alignment mapping qualities. Information on match, mismatch,
indel, strand, mapping quality and start and end of a read
are all encoded at the read base column. At this column, a
dot stands for a match to the reference base on the forward
strand, a comma for a match on the reverse strand, a
&rsquo;&gt;&rsquo; or &rsquo;&lt;&rsquo; for a reference
skip, &lsquo;ACGTN&rsquo; for a mismatch on the forward
strand and &lsquo;acgtn&rsquo; for a mismatch on the reverse
strand. A pattern &lsquo;\+[0-9]+[ACGTNacgtn]+&rsquo;
indicates there is an insertion between this reference
position and the next reference position. The length of the
insertion is given by the integer in the pattern, followed
by the inserted sequence. Similarly, a pattern
&lsquo;-[0-9]+[ACGTNacgtn]+&rsquo; represents a deletion
from the reference. The deleted bases will be presented as
&lsquo;*&rsquo; in the following lines. Also at the read
base column, a symbol &lsquo;^&rsquo; marks the start of a
read. The ASCII of the character following &lsquo;^&rsquo;
minus 33 gives the mapping quality. A symbol &lsquo;$&rsquo;
marks the end of a read segment.</p>

<p style="margin-left:26%; margin-top: 1em"><b>Input
Options: <br>
-6, --illumina1.3+</b></p>

<p style="margin-left:42%;">Assume the quality is in the
Illumina 1.3+ encoding.</p>

<p style="margin-left:26%;"><b>-A, --count-orphans</b></p>

<p style="margin-left:42%;">Do not skip anomalous read
pairs in variant calling.</p>


<p style="margin-left:26%;"><b>-b,&nbsp;--bam-list&nbsp;</b><i>FILE</i></p>

<p style="margin-left:42%;">List of input BAM files, one
file per line [null]</p>

<p style="margin-left:26%;"><b>-B, --no-BAQ</b></p>

<p style="margin-left:42%;">Disable probabilistic
realignment for the computation of base alignment quality
(BAQ). BAQ is the Phred-scaled probability of a read base
being misaligned. Applying this option greatly helps to
reduce false SNPs caused by misalignments.</p>


<p style="margin-left:26%;"><b>-C,&nbsp;--adjust-MQ&nbsp;</b><i>INT</i></p>

<p style="margin-left:42%;">Coefficient for downgrading
mapping quality for reads containing excessive mismatches.
Given a read with a phred-scaled probability q of being
generated from the mapped position, the new mapping quality
is about sqrt((INT-q)/INT)*INT. A zero value disables this
functionality; if enabled, the recommended value for BWA is
50. [0]</p>


<p style="margin-left:26%;"><b>-d,&nbsp;--max-depth&nbsp;</b><i>INT</i></p>

<p style="margin-left:42%;">At a position, read maximally
<i>INT</i> reads per input BAM. [250]</p>

<p style="margin-left:26%;"><b>-E, --redo-BAQ</b></p>

<p style="margin-left:42%;">Recalculate BAQ on the fly,
ignore existing BQ tags</p>


<p style="margin-left:26%;"><b>-f,&nbsp;--fasta-ref&nbsp;</b><i>FILE</i></p>

<p style="margin-left:42%;">The <b>faidx</b>-indexed
reference file in the FASTA format. The file can be
optionally compressed by <b>bgzip</b>. [null]</p>


<p style="margin-left:26%;"><b>-G,&nbsp;--exclude-RG&nbsp;</b><i>FILE</i></p>

<p style="margin-left:42%;">Exclude reads from readgroups
listed in FILE (one @RG-ID per line)</p>


<p style="margin-left:26%;"><b>-l,&nbsp;--positions&nbsp;</b><i>FILE</i></p>

<p style="margin-left:42%;">BED or position list file
containing a list of regions or sites where pileup or BCF
should be generated. If BED, positions are 0-based half-open
[null]</p>


<p style="margin-left:26%;"><b>-q,&nbsp;-min-MQ&nbsp;</b><i>INT</i></p>

<p style="margin-left:42%;">Minimum mapping quality for an
alignment to be used [0]</p>


<p style="margin-left:26%;"><b>-Q,&nbsp;--min-BQ&nbsp;</b><i>INT</i></p>

<p style="margin-left:42%;">Minimum base quality for a base
to be considered [13]</p>


<p style="margin-left:26%;"><b>-r,&nbsp;--region&nbsp;</b><i>STR</i></p>

<p style="margin-left:42%;">Only generate pileup in region.
Requires the BAM files to be indexed. If used in conjunction
with -l then considers the intersection of the two requests.
<i>STR</i> [all sites]</p>


<p style="margin-left:26%;"><b>-R,&nbsp;--ignore-RG</b></p>

<p style="margin-left:42%;">Ignore RG tags. Treat all reads
in one BAM as one sample.</p>


<p style="margin-left:26%;"><b>--rf,&nbsp;--incl-flags&nbsp;</b><i>STR|INT</i></p>

<p style="margin-left:42%;">Required flags: skip reads with
mask bits unset [null]</p>


<p style="margin-left:26%;"><b>--ff,&nbsp;--excl-flags&nbsp;</b><i>STR|INT</i></p>

<p style="margin-left:42%;">Filter flags: skip reads with
mask bits set [UNMAP,SECONDARY,QCFAIL,DUP]</p>


<p style="margin-left:26%;"><b>-x,&nbsp;--ignore-overlaps</b></p>

<p style="margin-left:42%;">Disable read-pair overlap
detection.</p>

<p style="margin-left:26%;"><b>Output Options: <br>
-o, --output</b> <i>FILE</i></p>

<p style="margin-left:42%;">Write pileup or VCF/BCF output
to <i>FILE</i>, rather than the default of standard
output.</p>

<p style="margin-left:42%; margin-top: 1em">(The same short
option is used for both <b>--open-prob</b> and
<b>--output</b>. If <b>-o</b>&rsquo;s argument contains any
non-digit characters other than a leading + or - sign, it is
interpreted as <b>--output</b>. Usually the filename
extension will take care of this, but to write to an
entirely numeric filename use <b>-o ./123</b> or <b>--output
123</b>.)</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="26%"></td>
<td width="14%">



<p style="margin-top: 1em" valign="top"><b>-g,&nbsp;--BCF</b></p> </td>
<td width="2%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Compute genotype
likelihoods and output them in the binary call format (BCF).
As of v1.0, this is BCF2 which is incompatible with the BCF1
format produced by previous (0.1.x) versions of
samtools.</p> </td>
<tr valign="top" align="left">
<td width="26%"></td>
<td width="14%">



<p style="margin-top: 1em" valign="top"><b>-v,&nbsp;--VCF</b></p> </td>
<td width="2%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Compute genotype
likelihoods and output them in the variant call format
(VCF). Output is bgzip-compressed VCF unless <b>-u</b>
option is set.</p></td>
</table>

<p style="margin-left:26%;"><b>Output Options for mpileup
format (without -g or -v): <br>
-O, --output-BP</b></p>

<p style="margin-left:42%;">Output base positions on
reads.</p>

<p style="margin-left:26%;"><b>-s, --output-MQ</b></p>

<p style="margin-left:42%;">Output mapping quality.</p>

<p style="margin-left:26%; margin-top: 1em"><b>Output
Options for VCF/BCF format (with -g or -v):</b></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="26%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top"><b>-D</b></p></td>
<td width="13%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Output per-sample
read depth [DEPRECATED - use <b>-t DP</b> instead]</p></td>
<tr valign="top" align="left">
<td width="26%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top"><b>-S</b></p></td>
<td width="13%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Output per-sample
Phred-scaled strand bias P-value [DEPRECATED - use <b>-t
SP</b> instead]</p></td>
</table>


<p style="margin-left:26%;"><b>-t,&nbsp;--output-tags&nbsp;</b><i>LIST</i></p>

<p style="margin-left:42%;">Comma-separated list of FORMAT
and INFO tags to output (case-insensitive): <b>DP</b>
(Number of high-quality bases, FORMAT), <b>DV</b> (Number of
high-quality non-reference bases, FORMAT), <b>DPR</b>
(Number of high-quality bases for each observed allele,
FORMAT), <b>INFO/DPR</b> (Number of high-quality bases for
each observed allele, INFO), <b>DP4</b> (Number of
high-quality ref-forward, ref-reverse, alt-forward and
alt-reverse bases, FORMAT), <b>SP</b> (Phred-scaled strand
bias P-value, FORMAT) [null]</p>


<p style="margin-left:26%;"><b>-u,&nbsp;--uncompressed</b></p>

<p style="margin-left:42%;">Generate uncompressed VCF/BCF
output, which is preferred for piping.</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="26%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top"><b>-V</b></p></td>
<td width="13%"></td>
<td width="58%">


<p style="margin-top: 1em" valign="top">Output per-sample
number of non-reference reads [DEPRECATED - use <b>-t DV</b>
instead]</p> </td>
</table>

<p style="margin-left:26%;"><b>Options for SNP/INDEL
Genotype Likelihood Computation (for -g <br>
or -v): <br>
-e,&nbsp;--ext-prob&nbsp;</b><i>INT</i></p>

<p style="margin-left:42%;">Phred-scaled gap extension
sequencing error probability. Reducing <i>INT</i> leads to
longer indels. [20]</p>


<p style="margin-left:26%;"><b>-F,&nbsp;--gap-frac&nbsp;</b><i>FLOAT</i></p>

<p style="margin-left:42%;">Minimum fraction of gapped
reads [0.002]</p>


<p style="margin-left:26%;"><b>-h,&nbsp;--tandem-qual&nbsp;</b><i>INT</i></p>

<p style="margin-left:42%;">Coefficient for modeling
homopolymer errors. Given an <i>l</i>-long homopolymer run,
the sequencing error of an indel of size <i>s</i> is modeled
as <i>INT</i>*<i>s</i>/<i>l</i>. [100]</p>

<p style="margin-left:26%;"><b>-I, --skip-indels</b></p>

<p style="margin-left:42%;">Do not perform INDEL
calling</p>


<p style="margin-left:26%;"><b>-L,&nbsp;--max-idepth&nbsp;</b><i>INT</i></p>

<p style="margin-left:42%;">Skip INDEL calling if the
average per-sample depth is above <i>INT</i>. [250]</p>


<p style="margin-left:26%;"><b>-m,&nbsp;--min-ireads&nbsp;</b><i>INT</i></p>

<p style="margin-left:42%;">Minimum number gapped reads for
indel candidates <i>INT</i>. [1]</p>


<p style="margin-left:26%;"><b>-o,&nbsp;--open-prob&nbsp;</b><i>INT</i></p>

<p style="margin-left:42%;">Phred-scaled gap open
sequencing error probability. Reducing <i>INT</i> leads to
more indel calls. [40]</p>

<p style="margin-left:42%; margin-top: 1em">(The same short
option is used for both <b>--open-prob</b> and
<b>--output</b>. When <b>-o</b>&rsquo;s argument contains
only an optional + or - sign followed by the digits 0 to 9,
it is interpreted as <b>--open-prob</b>.)</p>

<p style="margin-left:26%;"><b>-p, --per-sample-mF</b></p>

<p style="margin-left:42%;">Apply <b>-m</b> and <b>-F</b>
thresholds per sample to increase sensitivity of calling. By
default both options are applied to reads pooled from all
samples.</p>


<p style="margin-left:26%;"><b>-P,&nbsp;--platforms&nbsp;</b><i>STR</i></p>

<p style="margin-left:42%;">Comma-delimited list of
platforms (determined by <b>@RG-PL</b>) from which indel
candidates are obtained. It is recommended to collect indel
candidates from sequencing technologies that have low indel
error rate such as ILLUMINA. [all]</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="12%">



<p style="margin-top: 1em" valign="top"><b>reheader</b></p> </td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">samtools reheader
&lt;in.header.sam&gt; &lt;in.bam&gt;</p></td>
<td width="9%">
</td>
</table>

<p style="margin-left:26%; margin-top: 1em">Replace the
header in <i>in.bam</i> with the header in
<i>in.header.sam.</i> This command is much faster than
replacing the header with a BAM-&gt;SAM-&gt;BAM
conversion.</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="4%">


<p style="margin-top: 1em" valign="top"><b>cat</b></p></td>
<td width="11%"></td>
<td width="74%">


<p style="margin-top: 1em" valign="top">samtools cat [-h
header.sam] [-o out.bam] &lt;in1.bam&gt; &lt;in2.bam&gt; [
... ]</p></td>
</table>

<p style="margin-left:26%; margin-top: 1em">Concatenate
BAMs. The sequence dictionary of each input BAM must be
identical, although this command does not check this. This
command uses a similar trick to <b>reheader</b> which
enables fast BAM concatenation.</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="6%">


<p style="margin-top: 1em" valign="top"><b>sort</b></p></td>
<td width="9%"></td>
<td width="74%">


<p style="margin-top: 1em" valign="top">samtools sort
[<b>-l</b> <i>level</i>] [<b>-m</b> <i>maxMem</i>]
[<b>-o</b> <i>out.bam</i>] [<b>-O</b> <i>format</i>]
[<b>-n</b>] <b>-T</b> <i>out.prefix</i> [<b>-@</b>
<i>threads</i>] [<i>in.bam</i>]</p></td>
</table>

<p style="margin-left:26%; margin-top: 1em">Sort alignments
by leftmost coordinates, or by read name when <b>-n</b> is
used. An appropriate <b>@HD-SO</b> sort order header tag
will be added or an existing one updated if necessary.</p>

<p style="margin-left:26%; margin-top: 1em">The sorted
output is written to standard output by default, or to the
specified file (<i>out.bam</i>) when <b>-o</b> is used. This
command will also create temporary files
<i>out.prefix</i><b>.</b><i>%d</i><b>.bam</b> as needed when
the entire alignment data cannot fit into memory (as
controlled via the <b>-m</b> option).</p>


<p style="margin-left:26%; margin-top: 1em"><b>Options:</b></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="26%"></td>
<td width="14%">


<p style="margin-top: 1em" valign="top"><b>-l</b>
<i>INT</i></p> </td>
<td width="3%"></td>
<td width="57%">


<p style="margin-top: 1em" valign="top">Set the compression
level, from 0 to 9, for the final output file. If <b>-l</b>
is not used, the default compression level will apply.</p></td>
<tr valign="top" align="left">
<td width="26%"></td>
<td width="14%">


<p style="margin-top: 1em" valign="top"><b>-m</b>
<i>INT</i></p> </td>
<td width="3%"></td>
<td width="57%">


<p style="margin-top: 1em" valign="top">Approximately the
maximum required memory per thread, specified either in
bytes or with a <b>K</b>, <b>M</b>, or <b>G</b> suffix. [768
MiB]</p> </td>
<tr valign="top" align="left">
<td width="26%"></td>
<td width="14%">


<p style="margin-top: 1em" valign="top"><b>-n</b></p></td>
<td width="3%"></td>
<td width="57%">


<p style="margin-top: 1em" valign="top">Sort by read names
(i.e., the <b>QNAME</b> field) rather than by chromosomal
coordinates.</p> </td>
<tr valign="top" align="left">
<td width="26%"></td>
<td width="14%">


<p style="margin-top: 1em" valign="top"><b>-o</b>
<i>FILE</i></p> </td>
<td width="3%"></td>
<td width="57%">


<p style="margin-top: 1em" valign="top">Write the final
sorted output to <i>FILE</i>, rather than to standard
output.</p> </td>
<tr valign="top" align="left">
<td width="26%"></td>
<td width="14%">


<p style="margin-top: 1em" valign="top"><b>-O</b>
<i>FORMAT</i></p> </td>
<td width="3%"></td>
<td width="57%">


<p style="margin-top: 1em" valign="top">Write the final
output as <b>sam</b>, <b>bam</b>, or <b>cram</b>.</p></td>
</table>

<p style="margin-left:43%; margin-top: 1em">By default,
samtools tries to select a format based on the <b>-o</b>
filename extension; if output is to standard output or no
format can be deduced, <b>-O</b> must be used.</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="26%"></td>
<td width="14%">


<p style="margin-top: 1em" valign="top"><b>-T</b>
<i>PREFIX</i></p> </td>
<td width="3%"></td>
<td width="57%">


<p style="margin-top: 1em" valign="top">Write temporary
files to <i>PREFIX</i><b>.</b><i>nnnn</i><b>.bam.</b> This
option is required.</p></td>
<tr valign="top" align="left">
<td width="26%"></td>
<td width="14%">


<p style="margin-top: 1em" valign="top"><b>-@</b>
<i>INT</i></p> </td>
<td width="3%"></td>
<td width="57%">


<p style="margin-top: 1em" valign="top">Set number of
sorting and compression threads. By default, operation is
single-threaded.</p> </td>
</table>

<p style="margin-left:26%; margin-top: 1em">For
compatibility with existing scripts, <b>samtools sort</b>
also accepts the previous less flexible way of specifying
the final and temporary output filenames:</p>

<p style="margin-left:26%; margin-top: 1em">samtools sort
[<b>-nof</b>] [<b>-m</b> <i>maxMem</i>] <i>in.bam
out.prefix</i></p>

<p style="margin-left:26%; margin-top: 1em">The sorted BAM
output is written to <i>out.prefix</i><b>.bam</b> (or as
determined by the <b>-o</b> and <b>-f</b> options below) and
any temporary files are written alongside as
<i>out.prefix</i><b>.</b><i>%d</i><b>.bam.</b></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="26%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top"><b>-o</b></p></td>
<td width="5%"></td>
<td width="66%">


<p style="margin-top: 1em" valign="top">Output the final
alignment to the standard output.</p></td>
<tr valign="top" align="left">
<td width="26%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top"><b>-f</b></p></td>
<td width="5%"></td>
<td width="66%">


<p style="margin-top: 1em" valign="top">Use
<i>out.prefix</i> as the full output path and do not append
<b>.bam</b> suffix.</p></td>
</table>

<p style="margin-left:26%;"><b>-l</b>, <b>-m</b>,
<b>-n</b>, <b>-@</b></p>

<p style="margin-left:34%;">Accepted with the same meanings
as above.</p>

<p style="margin-left:26%; margin-top: 1em">This will
eventually be removed; you should move to using the more
flexible newer style of invocation.</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="7%">


<p style="margin-top: 1em" valign="top"><b>merge</b></p></td>
<td width="8%"></td>
<td width="74%">


<p style="margin-top: 1em" valign="top">samtools merge
[-nur1f] [-h inh.sam] [-R reg] [-b &lt;list&gt;]
&lt;out.bam&gt; &lt;in1.bam&gt; &lt;in2.bam&gt;
[&lt;in3.bam&gt; ... &lt;inN.bam&gt;]</p></td>
</table>

<p style="margin-left:26%; margin-top: 1em">Merge multiple
sorted alignment files, producing a single sorted output
file that contains all the input records and maintains the
existing sort order.</p>

<p style="margin-left:26%; margin-top: 1em">If <b>-h</b> is
specified the @SQ headers of input files will be merged into
the specified header, otherwise they will be merged into a
composite header created from the input headers. If in the
process of merging @SQ lines for coordinate sorted input
files, a conflict arises as to the order (for example
input1.bam has @SQ for a,b,c and input2.bam has b,a,c) then
the resulting output file will need to be re-sorted back
into coordinate order.</p>

<p style="margin-left:26%; margin-top: 1em">Unless the
<b>-c</b> or <b>-p</b> flags are specified then when merging
@RG and @PG records into the output header then any IDs
found to be duplicates of existing IDs in the output header
will have a suffix appended to them to diffientiate them
from similar header records from other files and the read
records will be updated to reflect this.</p>


<p style="margin-left:26%; margin-top: 1em"><b>OPTIONS:</b></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="26%">


<p style="margin-top: 1em" valign="top"><b>-1</b></p></td>
<td width="1%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Use zlib
compression level 1 to compress the output.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="26%">



<p style="margin-top: 1em" valign="top"><b>-b&nbsp;</b><i>FILE</i></p> </td>
<td width="1%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">List of input BAM
files, one file per line.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="26%">


<p style="margin-top: 1em" valign="top"><b>-f</b></p></td>
<td width="1%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Force to overwrite
the output file if present.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="26%">



<p style="margin-top: 1em" valign="top"><b>-h&nbsp;</b><i>FILE</i></p> </td>
<td width="1%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Use the lines of
<i>FILE</i> as &lsquo;@&rsquo; headers to be copied to
<i>out.bam</i>, replacing any header lines that would
otherwise be copied from <i>in1.bam</i>. (<i>FILE</i> is
actually in SAM format, though any alignment records it may
contain are ignored.)</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="26%">


<p style="margin-top: 1em" valign="top"><b>-n</b></p></td>
<td width="1%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">The input
alignments are sorted by read names rather than by
chromosomal coordinates</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="26%">



<p style="margin-top: 1em" valign="top"><b>-R&nbsp;</b><i>STR</i></p> </td>
<td width="1%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Merge files in the
specified region indicated by <i>STR</i> [null]</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="26%">


<p style="margin-top: 1em" valign="top"><b>-r</b></p></td>
<td width="1%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Attach an RG tag to
each alignment. The tag value is inferred from file
names.</p> </td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="26%">


<p style="margin-top: 1em" valign="top"><b>-u</b></p></td>
<td width="1%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Uncompressed BAM
output</p> </td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="26%">


<p style="margin-top: 1em" valign="top"><b>-c</b></p></td>
<td width="1%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Combine RG tags
with colliding IDs rather than adding a suffix to
differentiate them.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="26%">


<p style="margin-top: 1em" valign="top"><b>-p</b></p></td>
<td width="1%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Combine PG tags
with colliding IDs rather than adding a suffix to
differentiate them.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="26%">


<p><b>index</b></p></td>
<td width="1%"></td>
<td width="62%">
</td>
</table>

<p style="margin-left:11%;">samtools index [<b>-bc</b>]
[<b>-m</b> <i>INT</i>] <i>aln.bam</i>|<i>aln.cram</i></p>

<p style="margin-left:26%; margin-top: 1em">Index a
coordinate-sorted BAM or CRAM file for fast random access.
This index is needed when <i>region</i> arguments are used
to limit <b>samtools view</b> and similar commands to
particular regions of interest.</p>

<p style="margin-left:26%; margin-top: 1em">For a CRAM file
<i>aln.cram</i>, index file <i>aln.cram</i><b>.crai</b> will
be created. For a BAM file <i>aln.bam</i>, either
<i>aln.bam</i><b>.bai</b> or <i>aln.bam</i><b>.csi</b> will
be created, depending on the index format selected.</p>


<p style="margin-left:26%; margin-top: 1em"><b>Options:</b></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="24%">


<p style="margin-top: 1em" valign="top"><b>-b</b></p></td>
<td width="3%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Create a BAI index.
This is currently the default when no format options are
used.</p> </td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="24%">


<p style="margin-top: 1em" valign="top"><b>-c</b></p></td>
<td width="3%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Create a CSI index.
By default, the minimum interval size for the index is 2^14,
which is the same as the fixed value used by the BAI
format.</p> </td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="24%">


<p style="margin-top: 1em" valign="top"><b>-m</b>
<i>INT</i></p> </td>
<td width="3%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Create a CSI index,
with a minimum interval size of 2^INT.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="24%">


<p><b>idxstats</b></p></td>
<td width="3%"></td>
<td width="62%">
</td>
</table>

<p style="margin-left:11%;">samtools idxstats
&lt;aln.bam&gt;</p>

<p style="margin-left:26%; margin-top: 1em">Retrieve and
print stats in the index file. The output is TAB-delimited
with each line consisting of reference sequence name,
sequence length, # mapped reads and # unmapped reads.</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="7%">


<p style="margin-top: 1em" valign="top"><b>faidx</b></p></td>
<td width="8%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">samtools faidx
&lt;ref.fasta&gt; [region1 [...]]</p></td>
<td width="9%">
</td>
</table>

<p style="margin-left:26%; margin-top: 1em">Index reference
sequence in the FASTA format or extract subsequence from
indexed reference sequence. If no region is specified,
<b>faidx</b> will index the file and create
<i>&lt;ref.fasta&gt;.fai</i> on the disk. If regions are
specified, the subsequences will be retrieved and printed to
stdout in the FASTA format. The input file can be compressed
in the <b>BGZF</b> format.</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="11%">


<p style="margin-top: 1em" valign="top"><b>fixmate</b></p></td>
<td width="4%"></td>
<td width="74%">


<p style="margin-top: 1em" valign="top">samtools fixmate
[-rp] &lt;in.nameSrt.bam&gt; &lt;out.bam&gt;</p></td>
</table>

<p style="margin-left:26%; margin-top: 1em">Fill in mate
coordinates, ISIZE and mate related flags from a name-sorted
alignment.</p>


<p style="margin-left:26%; margin-top: 1em"><b>OPTIONS:</b></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="18%">


<p style="margin-top: 1em" valign="top"><b>-r</b></p></td>
<td width="9%"></td>
<td width="56%">


<p style="margin-top: 1em" valign="top">Remove secondary
and unmapped reads.</p></td>
<td width="6%">
</td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="18%">


<p style="margin-top: 1em" valign="top"><b>-p</b></p></td>
<td width="9%"></td>
<td width="56%">


<p style="margin-top: 1em" valign="top">Disable FR proper
pair check.</p></td>
<td width="6%">
</td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="18%">


<p><b>rmdup</b></p></td>
<td width="9%"></td>
<td width="56%"></td>
<td width="6%">
</td>
</table>

<p style="margin-left:11%;">samtools rmdup [-sS]
&lt;input.srt.bam&gt; &lt;out.bam&gt;</p>

<p style="margin-left:26%; margin-top: 1em">Remove
potential PCR duplicates: if multiple read pairs have
identical external coordinates, only retain the pair with
highest mapping quality. In the paired-end mode, this
command <b>ONLY</b> works with FR orientation and requires
ISIZE is correctly set. It does not work for unpaired reads
(e.g. two ends mapped to different chromosomes or orphan
reads).</p>


<p style="margin-left:26%; margin-top: 1em"><b>OPTIONS:</b></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="18%">


<p style="margin-top: 1em" valign="top"><b>-s</b></p></td>
<td width="9%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Remove duplicates
for single-end reads. By default, the command works for
paired-end reads only.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="18%">


<p style="margin-top: 1em" valign="top"><b>-S</b></p></td>
<td width="9%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Treat paired-end
reads and single-end reads.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="18%">


<p><b>calmd</b></p></td>
<td width="9%"></td>
<td width="62%">
</td>
</table>

<p style="margin-left:11%;">samtools calmd [-EeubSr] [-C
capQcoef] &lt;aln.bam&gt; &lt;ref.fasta&gt;</p>

<p style="margin-left:26%; margin-top: 1em">Generate the MD
tag. If the MD tag is already present, this command will
give a warning if the MD tag generated is different from the
existing tag. Output SAM by default.</p>


<p style="margin-left:26%; margin-top: 1em"><b>OPTIONS:</b></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="24%">


<p style="margin-top: 1em" valign="top"><b>-A</b></p></td>
<td width="3%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">When used jointly
with <b>-r</b> this option overwrites the original base
quality.</p> </td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="24%">


<p style="margin-top: 1em" valign="top"><b>-e</b></p></td>
<td width="3%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Convert a the read
base to = if it is identical to the aligned reference base.
Indel caller does not support the = bases at the moment.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="24%">


<p style="margin-top: 1em" valign="top"><b>-u</b></p></td>
<td width="3%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Output uncompressed
BAM</p> </td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="24%">


<p style="margin-top: 1em" valign="top"><b>-b</b></p></td>
<td width="3%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Output compressed
BAM</p> </td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="24%">


<p style="margin-top: 1em" valign="top"><b>-S</b></p></td>
<td width="3%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">The input is SAM
with header lines</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="24%">



<p style="margin-top: 1em" valign="top"><b>-C&nbsp;</b><i>INT</i></p> </td>
<td width="3%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Coefficient to cap
mapping quality of poorly mapped reads. See the
<b>pileup</b> command for details. [0]</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="24%">


<p style="margin-top: 1em" valign="top"><b>-r</b></p></td>
<td width="3%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Compute the BQ tag
(without -A) or cap base quality by BAQ (with -A).</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="24%">


<p style="margin-top: 1em" valign="top"><b>-E</b></p></td>
<td width="3%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Extended BAQ
calculation. This option trades specificity for sensitivity,
though the effect is minor.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="24%">


<p><b>targetcut</b></p></td>
<td width="3%"></td>
<td width="62%">
</td>
</table>

<p style="margin-left:11%;">samtools targetcut [-Q
minBaseQ] [-i inPenalty] [-0 em0] [-1 em1] [-2 em2] [-f ref]
&lt;in.bam&gt;</p>

<p style="margin-left:26%; margin-top: 1em">This command
identifies target regions by examining the continuity of
read depth, computes haploid consensus sequences of targets
and outputs a SAM with each sequence corresponding to a
target. When option <b>-f</b> is in use, BAQ will be
applied. This command is <b>only</b> designed for cutting
fosmid clones from fosmid pool sequencing [Ref. Kitzman et
al. (2010)].</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="7%">


<p style="margin-top: 1em" valign="top"><b>phase</b></p></td>
<td width="8%"></td>
<td width="74%">


<p style="margin-top: 1em" valign="top">samtools phase
[-AF] [-k len] [-b prefix] [-q minLOD] [-Q minBaseQ]
&lt;in.bam&gt;</p> </td>
</table>

<p style="margin-left:26%; margin-top: 1em">Call and phase
heterozygous SNPs.</p>


<p style="margin-left:26%; margin-top: 1em"><b>OPTIONS:</b></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="24%">


<p style="margin-top: 1em" valign="top"><b>-A</b></p></td>
<td width="3%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Drop reads with
ambiguous phase.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="24%">



<p style="margin-top: 1em" valign="top"><b>-b&nbsp;</b><i>STR</i></p> </td>
<td width="3%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Prefix of BAM
output. When this option is in use, phase-0 reads will be
saved in file <b>STR</b>.0.bam and phase-1 reads in
<b>STR</b>.1.bam. Phase unknown reads will be randomly
allocated to one of the two files. Chimeric reads with
switch errors will be saved in <b>STR</b>.chimeric.bam.
[null]</p> </td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="24%">


<p style="margin-top: 1em" valign="top"><b>-F</b></p></td>
<td width="3%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Do not attempt to
fix chimeric reads.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="24%">



<p style="margin-top: 1em" valign="top"><b>-k&nbsp;</b><i>INT</i></p> </td>
<td width="3%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Maximum length for
local phasing. [13]</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="24%">



<p style="margin-top: 1em" valign="top"><b>-q&nbsp;</b><i>INT</i></p> </td>
<td width="3%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Minimum
Phred-scaled LOD to call a heterozygote. [40]</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="24%">



<p style="margin-top: 1em" valign="top"><b>-Q&nbsp;</b><i>INT</i></p> </td>
<td width="3%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Minimum base
quality to be used in het calling. [13]</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="24%">


<p><b>flags</b></p></td>
<td width="3%"></td>
<td width="62%">
</td>
</table>

<p style="margin-left:11%;">samtools flags
INT|STR[,...]</p>

<p style="margin-left:26%; margin-top: 1em">Convert between
textual and numeric flag representation.</p>


<p style="margin-left:26%; margin-top: 1em"><b>FLAGS:</b></p>


<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="9%">


<p style="margin-top: 1em" valign="top"><b>bam2fq</b></p></td>
<td width="6%"></td>
<td width="71%">


<p style="margin-top: 1em" valign="top">samtools bam2fq
[-nO] [-s &lt;outSE.fq&gt;] &lt;in.bam&gt;</p></td>
<td width="3%">
</td>
</table>

<p style="margin-left:26%; margin-top: 1em">Converts a bam
into FASTQ format.</p>


<p style="margin-left:26%; margin-top: 1em"><b>OPTIONS:</b></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="26%"></td>
<td width="11%">


<p style="margin-top: 1em" valign="top"><b>-n</b></p></td>
<td width="1%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">By default, either
&rsquo;/1&rsquo; or &rsquo;/2&rsquo; is added to the end of
read names where the corresponding BAM_READ1 or BAM_READ2
flag is set. Using <b>-n</b> causes read names to be left as
they are.</p></td>
<tr valign="top" align="left">
<td width="26%"></td>
<td width="11%">


<p style="margin-top: 1em" valign="top"><b>-O</b></p></td>
<td width="1%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Use quality values
from OQ tags in preference to standard quality string if
available.</p> </td>
<tr valign="top" align="left">
<td width="26%"></td>
<td width="11%">


<p style="margin-top: 1em" valign="top"><b>-s FILE</b></p></td>
<td width="1%"></td>
<td width="62%">


<p style="margin-top: 1em" valign="top">Write singleton
reads in FASTQ format to FILE instead of outputting
them.</p> </td>
</table>


<p style="margin-left:11%;"><b>help</b>,&nbsp;<b>--help</b></p>

<p style="margin-left:26%;">Display a brief usage message
listing the samtools commands available. If the name of a
command is also given, e.g.,
<b>samtools&nbsp;help&nbsp;view</b>, the detailed usage
message for that particular command is displayed.</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="14%">



<p style="margin-top: 1em" valign="top"><b>--version</b></p> </td>
<td width="1%"></td>
<td width="74%">


<p style="margin-top: 1em" valign="top">Display the version
numbers and copyright information for samtools and the
important libraries used by samtools.</p></td>
</table>

<p style="margin-left:11%;"><b>--version-only</b></p>

<p style="margin-left:26%;">Display the full samtools
version number in a machine-readable format.</p>

<a name="REFERENCE SEQUENCES"></a>
<h2>REFERENCE SEQUENCES</h2>


<p style="margin-left:11%; margin-top: 1em">The CRAM format
requires use of a reference sequence for both reading and
writing.</p>

<p style="margin-left:11%; margin-top: 1em">When reading a
CRAM the <b>@SQ</b> headers are interrogated to identify the
reference sequence MD5sum (<b>M5:</b> tag) and the local
reference sequence filename (<b>UR:</b> tag). Note that
<i>http://</i> and <i>ftp://</i> based URLs in the UR: field
are not used, but local fasta filenames (with or without
<i>file://</i>) can be used.</p>

<p style="margin-left:11%; margin-top: 1em">To create a
CRAM the <b>@SQ</b> headers will also be read to identify
the reference sequences, but M5: and UR: tags may not be
present. In this case the <b>-T</b> and <b>-t</b> options of
samtools view may be used to specify the fasta or fasta.fai
filenames respectively (provided the .fasta.fai file is also
backed up by a .fasta file).</p>

<p style="margin-left:11%; margin-top: 1em">The search
order to obtain a reference is:</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">1.</p></td>
<td width="1%"></td>
<td width="85%">


<p style="margin-top: 1em" valign="top">Use any local file
specified by the command line options (eg -T).</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">2.</p></td>
<td width="1%"></td>
<td width="85%">


<p style="margin-top: 1em" valign="top">Look for MD5 via
REF_CACHE environment variable.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">3.</p></td>
<td width="1%"></td>
<td width="85%">


<p style="margin-top: 1em" valign="top">Look for MD5 in
each element of the REF_PATH environment variable.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">4.</p></td>
<td width="1%"></td>
<td width="85%">


<p style="margin-top: 1em" valign="top">Look for a local
file listed in the UR: header tag.</p></td>
</table>

<a name="ENVIRONMENT VARIABLES"></a>
<h2>ENVIRONMENT VARIABLES</h2>



<p style="margin-left:11%; margin-top: 1em"><b>REF_PATH</b></p>

<p style="margin-left:22%;">A colon separated (semi-colon
on Windows) list of locations in which to look for sequences
identified by their MD5sums. This can be either a list of
directories or URLs. Note that if a URL is included then the
colon in http:// and ftp:// and the optional port number
will be treated as part of the URL and not a PATH field
separator.</p>

<p style="margin-left:22%; margin-top: 1em">For URLs, the
text <b>%s</b> will be replaced by the MD5sum being read. If
no REF_PATH has been specified it will default to
<b>http://www.ebi.ac.uk/ena/cram/md5/%s</b></p>

<p style="margin-left:11%;"><b>REF_CACHE</b></p>

<p style="margin-left:22%;">This can be defined to a single
directory housing a local cache of references. Upon
downloading a reference it will be stored in the location
pointed to by REF_CACHE. When reading a reference it will be
looked for in this directory before searching REF_PATH. To
avoid many files being stored in the same directory, a
pathname may be constructed using %<i>num</i>s and %s
notation, consuming <i>num</i> characters of the MD5sum. For
example <b>/local/ref_cache/%2s/%2s/%s</b> will create 2
nested subdirectories with the filenames in the deepest
directory being the last 28 characters of the md5sum.</p>

<p style="margin-left:22%; margin-top: 1em">The REF_CACHE
directory will be searched for before attempting to load via
the REF_PATH search list.</p>

<p style="margin-left:22%; margin-top: 1em">To aid
population of the REF_CACHE directory a script
<b>misc/seq_ref_populate.pl</b> is provided in the Samtools
distribution. This takes a fasta file or a directory of
fasta files and generates the MD5sum named files.</p>

<a name="EXAMPLES"></a>
<h2>EXAMPLES</h2>


<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">o</p></td>
<td width="2%"></td>
<td width="86%">


<p style="margin-top: 1em" valign="top">Import SAM to BAM
when <b>@SQ</b> lines are present in the header:</p></td>
</table>

<p style="margin-left:14%; margin-top: 1em">samtools view
-bS aln.sam &gt; aln.bam</p>

<p style="margin-left:14%; margin-top: 1em">If <b>@SQ</b>
lines are absent:</p>

<p style="margin-left:14%; margin-top: 1em">samtools faidx
ref.fa <br>
samtools view -bt ref.fa.fai aln.sam &gt; aln.bam</p>

<p style="margin-left:14%; margin-top: 1em">where
<i>ref.fa.fai</i> is generated automatically by the
<b>faidx</b> command.</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">o</p></td>
<td width="2%"></td>
<td width="86%">


<p style="margin-top: 1em" valign="top">Convert a BAM file
to a CRAM file using a local reference sequence.</p></td>
</table>

<p style="margin-left:14%; margin-top: 1em">samtools view
-C -T ref.fa aln.bam &gt; aln.cram</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">o</p></td>
<td width="2%"></td>
<td width="77%">


<p style="margin-top: 1em" valign="top">Attach the
<b>RG</b> tag while merging sorted alignments:</p></td>
<td width="9%">
</td>
</table>

<p style="margin-left:14%; margin-top: 1em">perl -e
&rsquo;print
&quot;@RG\tID:ga\tSM:hs\tLB:ga\tPL:Illumina\n@RG\tID:454\tSM:hs\tLB:454\tPL:454\n&quot;&rsquo;
&gt; rg.txt <br>
samtools merge -rh rg.txt merged.bam ga.bam 454.bam</p>

<p style="margin-left:14%; margin-top: 1em">The value in a
<b>RG</b> tag is determined by the file name the read is
coming from. In this example, in the <i>merged.bam</i>,
reads from <i>ga.bam</i> will be attached <i>RG:Z:ga</i>,
while reads from <i>454.bam</i> will be attached
<i>RG:Z:454</i>.</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">o</p></td>
<td width="2%"></td>
<td width="41%">


<p style="margin-top: 1em" valign="top">Call SNPs and short
INDELs:</p> </td>
<td width="45%">
</td>
</table>

<p style="margin-left:14%; margin-top: 1em">samtools
mpileup -uf ref.fa aln.bam | bcftools call -mv &gt;
var.raw.vcf <br>
bcftools filter -s LowQual -e &rsquo;%QUAL&lt;20 ||
DP&gt;100&rsquo; var.raw.vcf &gt; var.flt.vcf</p>

<p style="margin-left:14%; margin-top: 1em">The <b>bcftools
filter</b> command marks low quality sites and sites with
the read depth exceeding a limit, which should be adjusted
to about twice the average read depth (bigger read depths
usually indicate problematic regions which are often
enriched for artefacts). One may consider to add <b>-C50</b>
to <b>mpileup</b> if mapping quality is overestimated for
reads containing excessive mismatches. Applying this option
usually helps <b>BWA-short</b> but may not other
mappers.</p>

<p style="margin-left:14%; margin-top: 1em">Individuals are
identified from the <b>SM</b> tags in the <b>@RG</b> header
lines. Individuals can be pooled in one alignment file; one
individual can also be separated into multiple files. The
<b>-P</b> option specifies that indel candidates should be
collected only from read groups with the <b>@RG-PL</b> tag
set to <i>ILLUMINA</i>. Collecting indel candidates from
reads sequenced by an indel-prone technology may affect the
performance of indel calling.</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">o</p></td>
<td width="2%"></td>
<td width="86%">


<p style="margin-top: 1em" valign="top">Generate the
consensus sequence for one diploid individual:</p></td>
</table>

<p style="margin-left:14%; margin-top: 1em">samtools
mpileup -uf ref.fa aln.bam | bcftools call -c | vcfutils.pl
vcf2fq &gt; cns.fq</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">o</p></td>
<td width="2%"></td>
<td width="32%">


<p style="margin-top: 1em" valign="top">Phase one
individual:</p> </td>
<td width="54%">
</td>
</table>

<p style="margin-left:14%; margin-top: 1em">samtools calmd
-AEur aln.bam ref.fa | samtools phase -b prefix - &gt;
phase.out</p>

<p style="margin-left:14%; margin-top: 1em">The
<b>calmd</b> command is used to reduce false heterozygotes
around INDELs.</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">o</p></td>
<td width="2%"></td>
<td width="75%">


<p style="margin-top: 1em" valign="top">Dump BAQ applied
alignment for other SNP callers:</p></td>
<td width="11%">
</td>
</table>

<p style="margin-left:14%; margin-top: 1em">samtools calmd
-bAr aln.bam &gt; aln.baq.bam</p>

<p style="margin-left:14%; margin-top: 1em">It adds and
corrects the <b>NM</b> and <b>MD</b> tags at the same time.
The <b>calmd</b> command also comes with the <b>-C</b>
option, the same as the one in <b>pileup</b> and
<b>mpileup</b>. Apply if it helps.</p>

<a name="LIMITATIONS"></a>
<h2>LIMITATIONS</h2>


<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="11%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">o</p></td>
<td width="2%"></td>
<td width="86%">


<p style="margin-top: 1em" valign="top">Unaligned words
used in bam_import.c, bam_endian.h, bam.c and bam_aux.c.</p></td>
<tr valign="top" align="left">
<td width="11%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">o</p></td>
<td width="2%"></td>
<td width="86%">


<p style="margin-top: 1em" valign="top">Samtools paired-end
rmdup does not work for unpaired reads (e.g. orphan reads or
ends mapped to different chromosomes). If this is a concern,
please use Picard&rsquo;s MarkDuplicates which correctly
handles these cases, although a little slower.</p></td>
</table>

<a name="AUTHOR"></a>
<h2>AUTHOR</h2>


<p style="margin-left:11%; margin-top: 1em">Heng Li from
the Sanger Institute wrote the C version of samtools. Bob
Handsaker from the Broad Institute implemented the BGZF
library. John Marshall and Petr Danecek contribute to the
source code and various people from the 1000 Genomes Project
have contributed to the SAM format specification.</p>

<a name="SEE ALSO"></a>
<h2>SEE ALSO</h2>



<p style="margin-left:11%; margin-top: 1em"><i>bcftools</i>(1),
<i>sam</i>(5)</p>

<p style="margin-left:11%; margin-top: 1em">Samtools
website: &lt;http://samtools.sourceforge.net&gt; <br>
Samtools latest source:
&lt;https://github.com/samtools/samtools&gt; <br>
File format specification of SAM/BAM,CRAM,VCF/BCF:
&lt;http://samtools.github.io/hts-specs&gt; <br>
HTSlib website: &lt;https://github.com/samtools/htslib&gt;
<br>
Bcftools website:
&lt;http://samtools.github.io/bcftools&gt;</p>
<hr>
