---
layout: default
title: Samtools - Documentation
---
			<a id="manuals"><h1>Man Pages</h1></a>
			<ul>
				<li>SAMtools</li>
				<li><a href="http://samtools.github.io/bcftools/bcftools.html">BCFtools</a></li>
				<li>HTSlib</li>
			</ul>
			<a id="specs"><h1>Specifications</h1></a>
			<p>
			SAMtools conforms to the specifications produced by the GA4GH File Formats working group. Details of the current specifications are available at: <a href="http://samtools.github.io/hts-specs">http://samtools.github.io/hts-specs</a>
			</p>
			<a id="pubs"><h1>Publications</h1></a>
			<ul>
			<li>
			Li H.*, Handsaker B.*, Wysoker A., Fennell T., Ruan J., Homer N., Marth G., Abecasis G., Durbin R. and 1000 Genome Project Data Processing Subgroup (2009) The Sequence alignment/map (SAM) format and SAMtools. Bioinformatics, 25, 2078-9. [PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/19505943">19505943</a>]
			 </li>
			<li>
			Li H. A statistical framework for SNP calling, mutation discovery, association mapping and population genetical parameter estimation from sequencing data. Bioinformatics. 2011 Nov 1;27(21):2987-93. Epub 2011 Sep 8. [PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/21903627">21903627</a>]
			</li>
			<li>Danecek P., Schiffels S., Durbin R. Multiallelic calling model in bcftools (-m) <a href="http://samtools.github.io/bcftools/call-m.pdf">http://samtools.github.io/bcftools/call-m.pdf</a></li>
			<li>Li H. Improving SNP discovery by base alignment quality. Bioinformatics. 2011 Apr 15;27(8):1157-8. doi: 10.1093/bioinformatics/btr076. Epub 2011 Feb 13. [PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/21320865">21320865</a>]</li>
			<li>Durbin R. Segregation based metric for variant call QC <a href="http://samtools.github.io/bcftools/rd-SegBias.pdf">http://samtools.github.io/bcftools/rd-SegBias.pdf</a></li>
			<li>Li H, Mathematical Notes on SAMtools Algorithms <a href="http://www.broadinstitute.org/gatk/media/docs/Samtools.pdf">http://www.broadinstitute.org/gatk/media/docs/Samtools.pdf</a></li>
			</ul>
