<html>
<head>
<title>Multisample SNP Calling</title>
<link rel="stylesheet" type="text/css" href="style.css" media="screen" />
</head>
<script language="JavaScript" src="jsInclude.js"></script>
<body onLoad="jsInclude('header', 'inc/header.html');">
<div id="wrap">
<span id="header"></span>

<div id="widemain">
<!-- BEGIN OF THE MAIN BODY  -->

<h2>Calling SNPs/INDELs with SAMtools/BCFtools</h2>

<h3>The basic Command line</h3>
<p>Suppose we have reference sequences in <i>ref.fa</i>, indexed
  by <b>samtools faidx</b>, and position sorted alignment
  files <i>aln1.bam</i> and <i>aln2.bam</i>, the following command lines
  call SNPs and short INDELs:
  <textarea name="code" class="c:nocontrols:nogutter" cols="60" rows="10">
  samtools mpileup -ugf ref.fa aln1.bam aln2.bam | bcftools view -bvcg - > var.raw.bcf
  bcftools view var.raw.bcf | vcfutils.pl varFilter -D100 > var.flt.vcf</textarea>

  where the <b>-D</b> option sets the maximum read depth to call a
  SNP. SAMtools acquires sample information from the <b>SM</b> tag in
  the <b>@RG</b> header lines. One alignment file can contain multiple
  samples; reads from one sample can also be distributed in different
  alignment files. SAMtools will regroup the reads anyway. In addition,
  if no <b>@RG</b> lines are present, each alignment file is taken as
  one sample.
</p>
<h3>Understanding the command line</h3>
<p>
  In the command line above, <b>samtools</b> collects summary
  information in the input BAMs, computes the likelihood of data given
  each possible genotype and stores the likelihoods in the BCF format
  (see below). It does not call variants.
</p>
<p>
  <b>Bcftools</b> applies the prior and does the actual calling. It can
  also concatenate BCF files, index BCFs for fast random access and
  convert BCF to VCF. In addition, <b>bcftools</b> can operate on some
  VCFs (e.g. calling SNPs from GL-tagged VCFs), but not for all VCFs;
  VCF to BCF conversion is not working at the moment, either.
</p>
<h3>Tuning the command line</h3>
<p>
  One should consider to apply the following parameters
  to <b>mpileup</b> in different scenarios:
  <ul>
	<li>Apply <b>-C50</b> to reduce the effect of reads with excessive
	  mismatches. This aims to fix overestimated mapping quality and
	  appears to be preferred for BWA-short.
	<li>Given multiple technologies, apply <b>-P</b> to specify which
	  technologies to use for collecting initial INDEL candidates. It is
	  recommended to find INDEL candidates from technologies with low
	  INDEL error rate, such as Illumina. When this option is in use,
	  the value(s) following the option must appear in the <b>PL</b> tag
	  in the <b>@RG</b> header lines.
	<li>Apply <b>-D</b> and <b>-S</b> to keep per-sample read depth and
	  strand bias. This is preferred if there are more than one samples
	  at high coverage.
	<li>Adjust <b>-m</b> and <b>-F</b> to control when to initiate indel
	  realignment (requiring r877+). Samtools only finds INDELs where
	  there are sufficient reads containing the INDEL at the same
	  position. It does this to avoid excessive realignment that is
	  computationally demanding. The default works well for many
	  low-coverage samples but not for, say, 500 exomes. In the latter
	  case, using <b>-m 3 -F 0.0002</b> (3 supporting reads at minimum
	  0.02% frequency) is necessary to find singletons.
	<li>Use `<b>-BQ0 -d10000000 -f ref.fa</b>' if the purpose is to get
	  the precise depth of coverage rather than call SNPs. Under this
	  setting, <b>mpileup</b> will count low-quality bases, process all
	  reads (by default the depth is capped at 8000), and skip the
	  time-demanding BAQ calculation.
	<li>Apply <b>-A</b> to use anomalous read pairs in <b>mpileup</b>,
	  which are not used by default (requring r874+).
  </ul>
</p>

<h3>Understanding the output: the VCF/BCF format</h3>
<h4>The VCF format</h4>
<p>
  The <a href="http://vcftools.sourceforge.net/specs.html">Variant
  Call Format</a> (VCF) is the emerging standard for storing variant
  data. Originally designed for SNPs and short INDELs, it also works for
  structural variations.
</p>
<p>
  VCF consists of a header section and a data section. The header must
  contain a line starting with one '#', showing the name of each field,
  and then the sample names starting at the 10th column. The data section
  is TAB delimited with each line consisting of at least 8 mandatory
  fields (the first 8 fields in the table below). The FORMAT field and
  sample information are allowed to be absent. We refer to the
  official <a href="http://vcftools.sourceforge.net/specs.html">VCF
  spec</a> for a more rigorous description of the format.
</p>
<table>
  <tr><th>Col<th>Field<th>Description</tr>
  <tr><td>1<td>CHROM<td>Chromosome name</tr>
  <tr><td>2<td>POS<td>1-based position. For an indel, this is the position preceding the indel.</tr>
  <tr><td>3<td>ID<td>Variant identifier. Usually the dbSNP rsID.</tr>
  <tr><td>4<td>REF<td>Reference sequence at POS involved in the variant. For a SNP, it is a single base.</tr>
  <tr><td>5<td>ALT<td>Comma delimited list of alternative seuqence(s).</tr>
  <tr><td>6<td>QUAL<td>Phred-scaled probability of all samples being homozygous reference.</tr>
  <tr><td>7<td>FILTER<td>Semicolon delimited list of filters that the variant fails to pass.</tr>
  <tr><td>8<td>INFO<td>Semicolon delimited list of variant information.</tr>
  <tr><td>9<td>FORMAT<td>Colon delimited list of the format of individual genotypes in the following fields.</tr>
  <tr><td>10+<td>Sample(s)<td>Individual genotype information defined by FORMAT.</tr>
</table>
<br>
<h4>The BCF Format</h4>
<p>
  BCF, or the binary variant call format, is the binary version of
  VCF. It keeps the same information in VCF, while much more efficient
  to process especially for many samples. The relationship between BCF
  and VCF is similar to that between BAM and SAM. The detailed format
  description of the BCF format can be found in <i>bcf.tex</i> included
  in the samtools source code package.
</p>
<h4>SAMtools/BCFtools specific information</h4>
<p>
  SAMtools/BCFtools may write the following tags in the <i>INFO</i> field in
  VCF/BCF.
</p>
<table>
  <tr><th>Tag<th>Description</tr>
  <tr><th>I16<td>16 integers:
		<table>
		  <tr><td>1<td>#reference Q13 bases on the forward strand
		  <td>2<td>#reference Q13 bases on the reverse strand
		  <tr><td>3<td>#non-ref Q13 bases on the forward strand
		  <td>4<td>#non-ref Q13 bases on the reverse strand
		  <tr><td>5<td>sum of reference base qualities
		  <td>6<td>sum of squares of reference base qualities
		  <tr><td>7<td>sum of non-ref base qualities
		  <td>8<td>sum of squares of non-ref base qualities
		  <tr><td>9<td>sum of ref mapping qualities
		  <td>10<td>sum of squares of ref mapping qualities
		  <tr><td>11<td>sum of non-ref mapping qualities
		  <td>12<td>sum of squares of non-ref mapping qualities
		  <tr><td>13<td>sum of tail distance for ref bases
		  <td>14<td>sum of squares of tail distance for ref bases
		  <tr><td>15<td>sum of tail distance for non-ref bases
		  <td>16<td>sum of squares of tail distance for non-ref
		</table></tr>
  <tr><th>INDEL<td>Indicating the variant is an INDEL.</tr>
  <tr><th>DP<td>The number of reads covering or bridging POS.</tr>
  <tr><th>DP4<td>Number of 1) forward ref alleles; 2) reverse ref; 3)
		forward non-ref; 4) reverse non-ref alleles, used in variant
		calling. Sum can be smaller than DP because low-quality bases are not counted.</tr>
  <tr><th>PV4<td>P-values for 1) strand bias (exact test); 2) baseQ bias
  (t-test); 3) mapQ bias (t); 4) tail distance bias (t)</tr>
  <tr><th>AF1<td>EM estimate of the site allele frequency of the strongest non-reference allele.</tr>
  <tr><th>CI95<td>Equal-tail (Bayesian) credible interval of the site allele frequency at the 95% level.</tr>
  <tr><th>HWE<td>P-value for Hardy-Weinberg equillibrium (chi-square test)</tr>
</table>
<ul>
  <li>For a read, suppose the k-th base is overlapping the reference
	base we are looking at. The tail distance for this read equals
	min{k-1,l-k} where l is the length of the read. If tail distance is
	small, the difference on the read may be caused by undetected
	indels.
  <li>In the BCF generated by SAMtools, an non-ref base 'X' represents
	an base has not been seen from the alignment data. Such a base is
	necessary to evaluate the probability of missing a non-ref allele
	due to sampling fluctuation.
  <li>SAMtools/BCFtools writes genotype likelihoods in the PL format
	which is a comma delimited list of phred-scaled data likelihoods of
	each possible genotype. For example, suppose REF=C and ALT=A,G,
	PL=7,0,13,37,40,49 means for the sample we are looking at,
	P(D|CC)=10^{-0.7}, P(D|CA)=1, P(D|CG)=10^{-1.3}, P(D|AA)=10^{-3.7},
	P(D|AG)=1e-4 and P(D|GG)=10^{-4.9}.
</ul>
<h2>Estimating the Allele Frequency Spectrum</h2>
<p>
  In addition to SNP/INDEL calling, <b>bcftools</b> is also able to
  estimate the Allele Frequency Spectrum (AFS) using the same model for
  variant calling.
</p>
<p>
  While calling SNPs, <b>bcftools</b> will print the estimated AFS to
  the error output at lines starting with <tt>[afs]</tt>. However, to
  accurately estimate AFS, we need to iterate the procedure. In most
  applications, we are only interested in the AFS conditional on a list
  of loci. Suppose we have the list of loci in file <i>cond.txt</i>
  with the first two columns in the file giving the coordinates, the
  procedure to estimate AFS is:
  <textarea name="code" class="c:nocontrols:nogutter" cols="60" rows="10">
  bcftools view -NIbl cond.txt data.bcf > cond.bcf
  bcftools view -cGP cond2 cond.bcf > round1.vcf 2> round1.afs
  bcftools view -cGP round1.afs cond.bcf > /dev/null 2> round2.afs
  bcftools view -cGP round2.afs cond.bcf > /dev/null 2> round3.afs
  ......</textarea>

  until the AFS converges, which usually takes less than 10 rounds of EM
  iterations. The first command line above extracts sites
  in <i>cond.txt</i> for efficiency in later steps. Option <b>-P</b>
  specifies the initial AFS (in SNP calling, this is prior), which can
  be a file (as in the 3rd and 4th command lines) or '<i>full</i>',
  '<i>cond2</i>' or '<i>flat</i>' (as in the 2nd command line). Choosing
  the right initial AFS helps accuracy and reduces iterations and
  potential overfitting.
</p>
<p>
  Another way to estimate AFS is to get the list of site allele
  frequency by
  <textarea name="code" class="c:nocontrols:nogutter" cols="60" rows="10">
  perl -ne 'print "$1\n" if /AF1=([^,;]+)/' round1.vcf</textarea> and
  then derive the histogram. It should be noted that AF1 does not use
  the initial AFS. No iterations are needed. However, the histogram of
  AF1 usually has a lot of noises. It is up to the users to decide which
  method to use.
</p>
<br>
<h2>Base Alignment Quality (BAQ)</h2>
<p>
  Base Alignment Quality (BAQ) is a new concept deployed in
  samtools-0.1.9+. It aims to provide an efficient and effective way to
  rule out false SNPs caused by nearby INDELs. The following shows the
  alignments of 6 reads by a typical read mapper in the presence of a
  4bp homozygous INDEL:
</p>
<pre>
<u> coor     12345678901234    5678901234567890123456 </u>
 ref      aggttttataaaac----aattaagtctacagagcaacta
<u> sample   aggttttataaaacAAATaattaagtctacagagcaacta </u>
 read1    aggttttataaaac****<u>aaAt</u>aa
 read2     ggttttataaaac****<u>aaAt</u>aaTt
 read3         ttataaaac<u>AAAT</u>aattaagtctaca
 read4             C<u>aaaT</u>****aattaagtctacagagcaac
 read5               <u>aaT</u>****aattaagtctacagagcaact
 read6                 <u>T</u>****aattaagtctacagagcaacta
</pre>
<p>
where capital bases represent differences from the reference and
underlined bases are the inserted bases. The alignments except for read3
are wrong because the 4bp insertion is misplaced. The mapper produces
such alignments because when doing a pairwise alignment, the mapper
prefers one or two mismatches over a 4bp insertion. What is hurting more
is that the wrong alignments lead to recurrent mismatches, which are
likely to deceive most site-independent SNP callers into calling false
SNPs.
</p>
<p>
  One way to avoid such an artifact is to do multi-sequence realignment,
  but the current implementations are very computationally
  demanding. SAMtools seeks another solution. It assigns each base a BAQ
  which is the Phred-scaled probability of the base being
  misaligned. BAQ is low if the base is aligned to a different reference
  base in a suboptimal alignment, and in this case a mismatch should
  contribute little to SNP calling even if the base quality is
  high. With BAQ, the mismatches in the example above are significantly
  downweighted. SAMtools will not call SNPs from that.
</p>
<p>
  The BAQ strategy is invoked by default in <b>mpileup</b>. To make
  other SNP callers take advantage of BAQ, one should run
  <textarea name="code" class="c:nocontrols:nogutter" cols="60" rows="10">
  samtools calmd -Abr aln.bam ref.fa > aln.baq.bam</textarea> to cap
  base quality by BAQ and then give <i>aln.baq.bam</i> to the SNP
  callers as the input. For high-coverage single-sample SNP calling, BAQ
  appears to be as effective as multi-sequence realignment, while being
  much faster and easier to use. Currently the BAQ strategy is the only
  practical way to avoid the INDEL artifact in low-coverage multi-sample
  SNP calling.
</p>
<br>

<h2>Limitations</h2>
<ul>
  <li>BCFtools does not properly handle multi-allelic variants. It only
	takes the strongest non-reference allele.
  <li>SAMtools does not properly compute the likelihoods of
	multi-allelic INDELs.
  <li>The VCF file produced by BCFtools does not strictly conform the
	VCF spec. For example, the GT genotype information is not always
	present because for the purpose of BCF, GT is unnecessary and takes
	disk space. In addition, GT is not the first as is required by the
	VCF spec. This can be fixed by the <b>bcf-fix.pl</b> script that
	comes with the source code package, and will be fixed in future
	(fixed in r880+).
</ul>
<!-- END OF THE MAIN BODY  -->
<link type="text/css" rel="stylesheet" href="SyntaxHighlighter.css"></link>
<script language="javascript" src="shCore.js"></script>
<script language="javascript" src="shBrushCpp.js"></script>
<script language="javascript">
  dp.SyntaxHighlighter.HighlightAll('code');
</script>
</div>

<div style="clear: both;"> </div>
<div id="footer">
<hr/>
<p style="text-align: right;">Last modified: 2010-12-10</p>
</div>
</div>
</body>
</html>
